ranger-tools/src/main/python/stress/stress-hbase-loadgenerator.py

```diff
@@ -0,0 +1,125 @@
+import subprocess
+import time
+import argparse
+import os
+from datetime import datetime
+
+def increase_memory_for_loadgenerator():
+    try:
+        cmd = "export HBASE_OPTS='-Xmx10g'"
+        print(cmd)
+        op = subprocess.call(cmd, shell=True)
+        print("Output:", op)
+    except subprocess.CalledProcessError as e:
+        print("Error in setting HBASE_HEAPSIZE:", e)
+        exit(1)
+def login(keytab_path, user_name):
+    try:
+        cmd = f"kinit -kt {keytab_path} {user_name}"
+        print(cmd)
+        login_op = subprocess.call(cmd, shell=True)
+        print("Login output:", login_op)
+    except subprocess.CalledProcessError as e:
+        print("Error in login:", e)
+        exit(1)
+
+def create_ltt_command_multiput(num_cols_per_cf=1000, num_threads=10, num_keys=100, table_name = "multitest",avg_data_size=2, num_col_families=3, col_family_pattern="cf%d", num_regions_per_server=1):
+    def get_column_families():
+        col_families = []
+        for i in range(num_col_families):
+            col_families.append(col_family_pattern % i)
+        return ','.join(col_families)
+    #Sample: hbase ltt -tn multitest -families f1,f2,f3 -write 20000:2:20 -multiput -num_keys 1000 -num_regions_per_server 1
+    cmd = f"hbase ltt -tn {table_name} -families {get_column_families()} -write {num_cols_per_cf}:{avg_data_size}:{num_threads}" \
+          f" -multiput -num_keys {num_keys} -num_regions_per_server {num_regions_per_server}"
+    return cmd
+
+
+def create_pe_command_multiget(multiget_batchsize=500, num_threads=10, num_keys=100, table_name="multitest", num_col_families=3):
+    #Sample: hbase pe --table=multitest --families=3 --columns=10000 --multiGet=10 --rows=1000 --nomapred randomRead 5
+
+    cmd = f"hbase pe --table={table_name} --families={num_col_families} --columns={num_cols_per_cf} " \
+          f"--multiGet={multiget_batchsize} --rows={num_keys} --nomapred randomRead {num_threads}"
+    return cmd
+
+
+
+def generate_hbase_load(op_type, multiget_batchsize, num_cf, num_keys_list, num_cols_per_cf, num_threads_list, metadata, csv_outfile="/root/ltt_output.csv", ):
```

Comment:
User might not have write permissions writing to the /root dir, please check.

Reply:
Command line argument -csv_output can be used in that case to provide a custom path where one has appropriate access